lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
```

Comment:
Good questions. I'm very unfamiliar with this project, and this change seemed to fix the bug, so I threw out the PR. I wish I knew the answers.